src/app/components/login/login.component.ts

```diff
@@ -43,8 +64,50 @@ export class LoginComponent {
     });
   }
 
-  encrypt(word: string) {
+  private getLoginParams(): User {
+    const username = this.userForm.value.username;
+    let usernameEncrypt = this.encrypt(username);
+
+    const password = this.userForm.value.password;
+    let passwordEncrypt = this.encrypt(password);
+
+    return { name: usernameEncrypt, password: passwordEncrypt };
+  }
+
+  private encrypt(word: string) {
     let byteword = new TextEncoder().encode(word);
     return btoa(String.fromCharCode(...new Uint8Array(byteword)));
   }
+
+  identifyErrorUser(error: any): boolean{
```

Comment:
esta función la podrías dejar en una sola pasando por parámetro el nombre del control para usarlo en this.userForm.controls['controlName']